marcdown_project/marcdown/api_views/notes.py

```diff
@@ -0,0 +1,129 @@
+from django.http import JsonResponse
+from rest_framework import viewsets
+from django.shortcuts import get_object_or_404
+from rest_framework.renderers import JSONRenderer
+
+from rest_framework.decorators import action
+
+from django.contrib.auth.models import User
+from marcdown.models import Note, Profile
+from marcdown.serializers import NoteSerializer
+
+class NoteViewSet(viewsets.ViewSet):
+    # get
+    def retrieve(self, request, pk=None):
+        '''
+        Gets the note corresponding to the given id,
+        and responds with its description and content
+        '''
+        queryset = Note.objects.all()
+        note = get_object_or_404(queryset, id=pk)
+        serializer = NoteSerializer(note)
+        return JsonResponse(serializer.data)
+    
+    # post (public = True, readonly = False, sharers = [])
+    # needs auth
+    def create(self, request):
+        '''
+        Creates an empty note with the given parameters
+        '''
+        user = request.user
+        if user.is_authenticated:
+            data = request.data
+            note = Note(
+                owner=user.profile,
+                public=getattr(data, "public", True),
+                read_only=getattr(data, "readOnly", False),
+                title="Blank",
+                content="",
```

Comment:
Is it possible to set the note content and title when creating it?

Reply:
It certainly is, I just figured it wouldn't be used.

It won't take much work to add it, so I'll do it.

Reply:
Also, the title is just the first-level header, so it can't be set.